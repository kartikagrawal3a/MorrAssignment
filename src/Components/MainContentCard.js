import React from 'react';
import MainContentCSS from '../styles/MainContent.module.css';

const MainContentCard = (props) => {
    return(
        <>  <div title="main">
                <div className={MainContentCSS.card} title="card">
                    <img src={props.url} alt="content_pic" title="cardImg" />
                </div>
                <p className="info" title="title">{props.title}</p>
            </div>
        </>
    );
}

export default MainContentCard;