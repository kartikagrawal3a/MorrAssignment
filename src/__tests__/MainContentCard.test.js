import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import MainContentCard from '../Components/MainContentCard';

afterEach(()=>{
    cleanup();
})

test("checkHeaderRender1", ()=>{
    const cat = "Ages of Love";
    const { getByTitle} = render(<MainContentCard title = {cat} />);
    getByTitle("main");
    getByTitle("card");
    getByTitle("cardImg")
    const title = getByTitle("title");
    expect(title).toHaveTextContent("Ages of Love")
})

test("checkHeaderRender2", ()=>{
    const cat = "Danny Collins";
    const { getByTitle} = render(<MainContentCard title = {cat} />);
    getByTitle("main");
    getByTitle("card");
    getByTitle("cardImg")
    const title = getByTitle("title");
    expect(title).toHaveTextContent("Danny Collins")
})

test('match snapshot #1', ()=>{
    const cat = "Ages of Love";
    const tree = renderer.create(<MainContentCard type = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('match snapshot #2', ()=>{
    const cat = "Danny Collins";
    const tree = renderer.create(<MainContentCard type = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})