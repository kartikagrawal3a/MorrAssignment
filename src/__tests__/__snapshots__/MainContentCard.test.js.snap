// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot #1 1`] = `
Array [
  "  ",
  <div
    title="main"
  >
    <div
      className="card"
      title="card"
    >
      <img
        alt="content_pic"
        title="cardImg"
      />
    </div>
    <p
      className="info"
      title="title"
    />
  </div>,
]
`;

exports[`match snapshot #2 1`] = `
Array [
  "  ",
  <div
    title="main"
  >
    <div
      className="card"
      title="card"
    >
      <img
        alt="content_pic"
        title="cardImg"
      />
    </div>
    <p
      className="info"
      title="title"
    />
  </div>,
]
`;
